python/GafferDelightUI/ShaderMenu.py

```diff
@@ -71,35 +71,158 @@ def __shaderSubMenu( oslPath ) :
+				subMenu = "Utility"
+			elif shaderFileName == "dlAOVGroup" :
+				subMenu = "AOV"
+			elif shaderFileName == "dlAOVGroupFour" :
+				subMenu = "AOV"
+			elif shaderFileName == "hsvToRgb" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "rgbToHsv" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "blendColors" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "Curve UV Coordinates" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "uvCoord" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "contrast" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "remapHsv" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "remapValue" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "shadingEngine_surface_switch" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "shadingEngine_displacement_switch" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "uvChooser" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "multiplyDivide" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "Set Range" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "reverse" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "uvCombine" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "colorCombine" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "luminance" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "smear" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "remapColor" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "vectorProduct" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "fourByFourMatrix" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "uvCoordEnvironment" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "channels" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "samplerInfo" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "distanceBetween" :
+				subMenu = "Utility (Legacy)"
+			elif shaderFileName == "ocean" :
+				subMenu = "Texture 2D (Legacy)"
+			elif shaderFileName == "psdFileTex" :
+				subMenu = "Texture 2D (Legacy)"
+			elif shaderFileName == "file" :
+				subMenu = "Texture 2D (Legacy)"
+			elif shaderFileName == "place2dTexture" :
+				subMenu = "Texture 2D (Legacy)"
+			elif shaderFileName == "ramp" :
+				subMenu = "Texture 2D (Legacy)"
+			elif shaderFileName == "imagePlane" :
+				subMenu = "Texture 2D (Legacy)"
+			elif shaderFileName == "stencil" :
+				subMenu = "Texture 2D (Legacy)"
+			elif shaderFileName == "place3dTexture" :
+				subMenu = "Texture 3D (Legacy)"
+			elif shaderFileName == "stucco" :
+				subMenu = "Texture 3D (Legacy)"
+			elif shaderFileName == "envChrome" :
+				subMenu = "Texture 3D (Legacy)"
+			elif shaderFileName == "snow" :
+				subMenu = "Texture 3D (Legacy)"
```

Comment:
Similarly here, it might be more readable to do something like
```
subMenu = {
    "shadingEngine_displacement": "Displacement",
    "builtinDisplacement", "Displacement",
    ...
}.get( shaderFileName, None )
```

I probably should have done that myself in the earlier commit. But if you prefer the conditionals I won't hold up merging.

Reply:
Thanks for this suggestion as well! Definitely looks better. I had to shift around some of the surrounding code since the above approach needs to be done before querying the tags and building the additional submenus from them, but I think the code ended up nicely organized at the end.